### DISCUSSION TITLE
Save sublayer edit target in Maya file, plus layer locking?

### DISCUSSION THREAD

**Discussion by daviddeacon** [Autodesk/maya-usd#1893] [Ideas] [answered]
Hi,

I'm working on a shot based workflow, and setting up sublayers for department/artist etc. I need this to be as artist friendly as possible.

I want to pre setup Maya scenes for each department/artist, and I want the edit target layer to be preset but this isn't being saved in the Maya scene and is reverting to the top layer on scene load/reload. Is it possible to add this to the Maya scene data?

On a related point, it's very easy for artist to switch to another department/artist layer and create all sorts of edits. It would be great if you could add a layer locking mechanism. I imagine a little padlock icon on each layer next to the mute button, ideally again this attribute would be saved in the Maya scene data.

Thanks,
David.

**Comment by brycegbrazen** [comment]
Flagging this as a request from myself as well. Curious if there's any development on this workflow?

**Comment by womanyoyoyo-adsk** [comment, marked as answer]
Hi,

We have the editTarget issue internally logged. We haven't started development on it yet but we're seeing that this is very blocking indeed. Thanks for flagging this and explaining your workflow. 

Question about locking layers. Would you expect this Maya layer locking mechanism to be separate from USD's sdf.layer.SetPermissionToEdit? I am thinking in Maya we show sdf.layer.SetPermissionToEdit=false as read-only layers and locking layers would be a separate, only-Maya mechanism (a lock icon for example) that is not authored in the USD data model. Thoughts?

https://graphics.pixar.com/usd/release/api/class_sdf_layer.html#a32ad22bde9522ec46ef46ce2b88dfd14

**Comment by daviddeacon** [comment]
Hi,

Thanks for confirming this has been logged internally.

Layer locking would ideally be part of the Maya data model. As we would want an artist to be able to unlock their target layer. If the lock was part of the USD stage, it would unlock for everyone that is working on the same shot.

**Reply by brycegbrazen** [reply]
+1 here.

**Comment by womanyoyoyo-adsk** [comment]
Edit target to persist when saving the Maya file - internal issue [MAYA-114188]

**Comment by womanyoyoyo-adsk** [comment]
The lock on the the USD stage - is its purpose to unlock/lock all the layers in a stage in bulk (quickly)? Or is there another purpose.

**Reply by daviddeacon** [reply]
That would be a useful help, one thing that would be very important is being able to lock/unlock layers via python. So I can programatically setup scenes.

**Comment by womanyoyoyo-adsk** [comment]
Also mentioned here: #2556

**Comment by Aquamenti9** [comment]
Hello,
I'm interested by this layer locking feature as well!
Also, looking into "sdf.layer.SetPermissionToEdit" I couldn't find a way to save the permission modification into the stage or the usds, is it something that can only by edited temporarily in maya? Or is there another way to setup a stage with some uneditable layer and keep that information when opening the stage later on?
Thanks!

**Reply by womanyoyoyo-adsk** [reply]
Currently Maya does not author the layer's SetPermissionToEdit metadata. Maya respects this and visualizes a layer who's metadata is setPermissionToEdit=false by showing the layer as dimmed and the editTarget is disabled. There is no way currently in Maya to have this information be saved by the stage.

**Comment by scottrenaud** [comment]
Hey @daviddeacon and @brycegbrazen, 

We posted [Feedback needed on new UI - Layer Locking #3486](https://github.com/Autodesk/maya-usd/discussions/3486) and would love your feedback on the designs if you have a chance!

Thanks!
Scott
